### PR TITLE
feat(tokens): display mainnet tokens total USD

### PR DESCRIPTION
**Pull request by DenysKarmazynDFINITY**
# Motivation

The goal is to display mainnet tokens total USD balances in `NetworkSwitscher`.

Screenshot 2024-09-06 at 17 00 56

### CODE REVIEW

File: src/frontend/src/lib/components/networks/MainnetNetwork.svelte [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+	import NetworkComponent from '$lib/components/networks/Network.svelte';
+	import { enabledMainnetTokensUsdBalancesPerNetwork } from '$lib/derived/tokens.derived';
+
+	export let network: Network;
+
+	let totalUsd: number;
+	$: totalUsd = $enabledMainnetTokensUsdBalancesPerNetwork[network.id];
+</script>
+
+<NetworkComponent {network} {totalUsd} on:icSelected />
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/lib/components/networks/MainnetNetwork.svelte:12]: I would rename it `usdBalance` just to have the same convention

File: src/frontend/src/lib/components/networks/Network.svelte [hunk 2]
```diff
@@ -3,11 +3,12 @@
...
+	export let totalUsd: number | undefined = undefined;
 
 	let id: NetworkId;
 	let name: string;
 	let icon: string | undefined;
 	$: ({ id, name, icon } = network);
 </script>
 
-<NetworkButton {id} {name} {icon} on:icSelected />
+<NetworkButton {id} {name} {totalUsd} {icon} on:icSelected />
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/lib/components/networks/Network.svelte:14]: Same here

File: src/frontend/src/lib/components/networks/NetworkButton.svelte [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
 	import { networkId } from '$lib/derived/network.derived';
 	import type { NetworkId } from '$lib/types/network';
+	import { formatUSD } from '$lib/utils/format.utils';
 	import { gotoReplaceRoot, isRouteTransactions, switchNetwork } from '$lib/utils/nav.utils';
 
 	export let id: NetworkId | undefined;
 	export let name: string;
 	export let icon: string | undefined;
-	export let description: string | undefined = undefined;
+	export let totalUsd: number | undefined = undefined;
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/lib/components/networks/NetworkButton.svelte:16]: Ditto

File: src/frontend/src/lib/components/networks/NetworkButton.svelte [hunk 4]
```diff
@@ -28,7 +30,12 @@
 </script>
 
 <button class="w-full flex justify-between" on:click={onClick}>
-	<TextWithLogo {name} {description} {icon} logo="start" />
+	<TextWithLogo
+		{name}
+		{icon}
+		logo="start"
+		description={nonNullish(totalUsd) ? formatUSD(totalUsd) : undefined}
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/lib/components/networks/NetworkButton.svelte:37]: I never remember: is this reactive? Meaning, if balance changes in the parent, it changes here to right?

File: src/frontend/src/lib/components/networks/NetworksSwitcher.svelte [hunk 5]
```diff
@@ -23,6 +25,12 @@
 
 	let testnets: boolean;
 	$: testnets = $testnetsStore?.enabled ?? false;
+
+	let mainnetTokensTotalUsd: number;
+	$: mainnetTokensTotalUsd = $networksMainnets.reduce(
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/lib/components/networks/NetworksSwitcher.svelte:30]: usdBalance?

File: src/frontend/src/lib/components/networks/NetworkButton.svelte (hunk 4 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/components/networks/NetworkButton.svelte:37]: Yes, it re-renders if a prop changes

File: src/frontend/src/lib/components/networks/MainnetNetwork.svelte (hunk 1 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/components/networks/MainnetNetwork.svelte]: Good point, renamed

**Review by AntonioVentilii-DFINITY — APPROVED** [review]
LGTM! 

Before merge, can you put the image of how it is now in the PR description?

File: src/frontend/src/lib/components/networks/NetworkButton.svelte [hunk 6]
```diff
@@ -27,8 +29,13 @@
 	};
 </script>
 
-<button class="w-full flex justify-between" on:click={onClick}>
-	<TextWithLogo {name} {description} {icon} logo="start" />
+<button class="w-full flex justify-between items-start" on:click={onClick}>
```
Comment (DenysKarmazynDFINITY) [inline src/frontend/src/lib/components/networks/NetworkButton.svelte:32]: @peterpeterparker in one of the previous PRs, we removed `items-start` from this line, but it appears there is a global `button.scss` file which adds `align-items: center` to all `button`s. In this particular case, I need to have `align-items: flex-start`, so I added back that class.

**Pull request merged**